README.md

```diff
@@ -213,6 +213,20 @@ To run the Streisand provisioning against an existing server:
         --ssh-user root \
         --site-config global_vars/noninteractive/digitalocean-site.yml
 
+Docker Installation
+-------------------
+
+If you don't want to pollute your local environment, you can use docker.
+Just build the image, then run it with your ssh key and output folder
+mounted:
+
+    # build
+    docker build . -t streisand
+
+    # run
+    docker run -it -v ~/.ssh:/root/.ssh -v $PWD/generated-docs:/root/generated-docs streisand:latest
```

Comment:
This is much cleaner than what I did, although you'll get the same pushback on `~/.ssh` mounting (don't expose all the user's keys). Until the two of us come up with something clever or cleaner to do about keys, we should at least mount them read-only. 

Come to think of it, we can make the user's choice of SSH keys *appear* at `/root/.ssh/id_rsa`, through (read-only) bind mounting. This might resolve @cpu's issues with mounting the whole `~/.ssh`.

I think we need to at least do the run phase in a script, in order to validate a few things. For starters, we do need to pick an SSH keypair that exists, or else Docker might do something really *dumb* with that bind. (The directory `/path/to/conf` exists on my machine, because Docker was happy to create the directory mentioned in the cut&pasted line. :‐(

Reply:
I suggest that we add a step in the `Dockerfile` to generate a brand new key pair, then place it in the `generated_docs` directory. The user is then free to copy it back to his `~/.ssh` or use it from there using the `-i` parameter.

Reply:
What I *really* want to do is use the ssh-agent to hold the private keys, and extract the public key from the agent for an `id_rsa.pub`. My primary SSH key is held on a Yubikey, and it simply does not *have* an `id_rsa`.

Generating the `id_rsa` key is great for people who don't do this very much. But I have about a zillion machines I need to log into, and every additional `id_rsa` means more complexity for me--I'm already at the five cert limit, so I need to add `IdentitiesOnly` clauses to `~/.ssh/config` to pick which key goes where.

Another user's requirement was solved in Streisand in a way that made using `ssh-agent` harder, so this has been on my back burner.

I think the best thing is to read-only bind-mount the keys specified on the host to `/root/.ssh/id_rsa*` `id_rsa` and `id_rsa.pub` in. This exposes the private `id_rsa` to the container, but without the agent, that's inevitable. :‐(